executor/calibrate_resource.go

```diff
@@ -334,17 +384,22 @@ func getNumberFromMetrics(ctx context.Context, exec sqlexec.RestrictedSQLExecuto
 	return rows[0].GetFloat64(0), nil
 }
 
-func getValuesFromMetrics(ctx context.Context, exec sqlexec.RestrictedSQLExecutor, query, metrics string) ([]float64, error) {
+func getValuesFromMetrics(ctx context.Context, sctx sessionctx.Context, exec sqlexec.RestrictedSQLExecutor, query, metrics string) (*timeSeriesValues, error) {
 	rows, _, err := exec.ExecRestrictedSQL(ctx, []sqlexec.OptionFuncAlias{sqlexec.ExecOptionUseCurSession}, query)
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
 	if len(rows) == 0 {
 		return nil, errors.Errorf("metrics '%s' is empty", metrics)
 	}
-	ret := make([]float64, 0, len(rows))
+	ret := make([]*timePointValue, 0, len(rows))
 	for _, row := range rows {
-		ret = append(ret, row.GetFloat64(0))
+		if tp, err := row.GetTime(0).AdjustedGoTime(sctx.GetSessionVars().Location()); err == nil {
+			ret = append(ret, &timePointValue{
+				tp:  tp,
+				val: row.GetFloat64(1),
```

Comment:
It was wrong before, why is it 1？

Reply:
The expression is changed, the original expr only returns the value, the new expr returns the time and value